### ISSUE TITLE
[RFE] Prioritize PR/s for merge

### ISSUE THREAD

**Issue by EdDev** [kubevirt/kubevirt#5202]
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind enhancement

**What happened**:
There are scenarios where specific PR/s are required to urgently get merged, solving an critical bug or fixing flakiness that blocks all CI jobs.
Non prioritized PR/s will compete on getting merged and may delay the most important PR/s.

This has been seen recently with the test falkes on CI, where PR/s with low priority get merged before to high priority ones.

**What you expected to happen**:
Once a PR is recognized to have high priority (due to a fix that will resolve blocking CI issues, urgent user bugs or a feature that must be included in the next release), it should get prioritized to be merged.

**Anything else we need to know?**:

**Environment**:
- KubeVirt version (use `virtctl version`):
- Kubernetes version (use `kubectl version`):
- VM or VMI specifications:
- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by EdDev** [comment]
One idea to implement this is using GitHub Actions, monitoring a "priority-merge" label on PR/s and commenting other PR/s which do not have this label with a `/hold` and adding a label which mentions this has been done.
Once no PR/s have the "priority-merge" label, issue a `/hold cancel` on all PR/s that have been hold previously.

**Comment by jmguzik** [comment]
@EdDev I asked this some time ago and I do not know if this is the right place to ask it again but have you thought of actually splitting the test plans to make them pass in smaller batches?
It is a real pity and a waste of time to see the whole 4.5h test plan failing because one of the tests is failing (1 from 700ish). Maybe it would be worthwhile to split 1 test plan into 4 smaller ones? To not waste resources they could be run on a limited number of jobs or one after the other.

I really do not know your infra, so please forgive if that sounds odd, but I have to be honest here - the CI is a major obstacle if you want to do something in kubevirt as a community member. I sometimes retest for 3-4 times x 4h to get this one (and next time the other) test green again...

**Comment by EdDev** [comment]
@jmguzik, work is in progress to stabilize the CI. Splitting into several jobs with other interesting options has been proposed.
There is no full consensus on how to do this and its cost vs gain is not trivial.

I hope a proposal will land soon so we can discuss it in more details.

In the meantime, this request is about controlling PR merges in extreme cases, something we are lacking today and makes it hard to recover faster.

**Comment by EdDev** [comment]
During the last community meeting (from 17.03.2021) @rmohr voted to drop this request in favor of maintainers forcibly merge PR/s which are prioritized.

A practical case where such a feature was needed occurred only once, when no maintainer was available.
I am ok with deferring/closing this but please note that:
- A systematic workflow is always preferred over a force action which requires special care (to check that the CI jobs are good, not to race with another merge that happened 2sec before).
- Maintainers are not always available, especially in regards to working days.
- I'm not sure if all maintainers will take this risk of forcebly merging PRs or if the have the ability to do so on this project (i.e. an approver may not have write/admin access to the repo).

**Comment by phoracek** [comment]
Thanks @EdDev. Closing for now. Let's revisit this if the problem reoccurs more often.
